ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/util/DataColumnSidecarELRecoveryManagerImpl.java

```diff
@@ -332,12 +339,11 @@ private void onFirstSeen(
     }
 
     asyncRunner
-        .runAsync(
-            () ->
-                // fetch blobs from EL with no delay
-                fetchMissingBlobsFromLocalEL(slotAndBlockRoot))
-        .handleException(this::logLocalElBlobsLookupFailure)
-        .ifExceptionGetsHereRaiseABug();
+        .runWithRetry(
```

Comment:
Before change we were not expecting recovery started after receiving all dataColumnSidecars through gossip because we start on first sidecar (or even block) and require at least 4. We were not able to get other sidecars immediately. But now as we wait 500ms for retry, next retry could happen after receiving all required sidecars by gossip so we should stop retrying. 
TLDR: check that we have not already received sidecars in custody and sampler before next retry. The simplest way is to catch it all here as far as we already subscried onNewValidSidecar

Reply:
As discussed, added an index tracking to check if we haven't already received the missing columns before retrying fetching them from the EL